feat: 공고 상세 페이지 UI 및 일부 기능 구현

KoCeleste opened the pull request:
<!-- PR 제목 규칙:
[type]: 작업 요약 (#이슈번호)
예시: feat: 로그인 페이지 UI 구현 (#12)
-->

## #️⃣ 연관된 이슈

- Resolves #67, #84

## 📑 구현 사항

- 공고 상세 조회 MSW 핸들러 및 API 함수 구현
- 공고 상세 페이지 컴포넌트 구현 (DetailHeader, DetailInfo, DetailContent, DetailAction)
- 라우팅 설정 (/recruitments/:id)
- 로딩 및 에러 상태 처리
- 목록으로 돌아가는 버튼 구현
- 반응형 디자인 적용
- zod 라이브러리 추가 (데이터 유효성 검증용)

## 🌀 어려웠던 점 / 고민했던 부분

- MSW 핸들러에서 동적 라우팅 파라미터(:id) 처리 방법
- 상세 페이지 컴포넌트 분리 기준 (재사용성 vs 가독성)

## 📸 구현 이미지

- 

https://github.com/user-attachments/assets/ae9ce428-f376-4672-832a-a780ea3df767
![로그인1](https://github.com/user-attachments/assets/17d49ccc-fa79-4f20-8e94-b994b8999f65)
![로그인2](https://github.com/user-attachments/assets/90d89eb1-7103-4270-8924-c7b080525b60)


## ❇️ 코드 설명

- `useRecruitmentDetail` 커스텀 훅으로 데이터 페칭 로직 분리
- DetailHeader/DetailInfo/DetailContent/DetailAction로 컴포넌트 역할 분리
- React Router의 useParams로 동적 라우팅 처리

## 💬 리뷰 요청사항

> 리뷰어에게 피드백을 받고 싶은 지점이 있다면 작성해주세요.

- 이번 이슈에는 작업하다보니 컴포넌트 분리를 하면서 테스트 겸 기능을 넣고 빼고 하면서 (기능 부분 코드를 미처 다 못빼고 작업하고 돌아가고 했던 부분이 잦았어서) 신경을 잘 못쓰고 한꺼번에 추가하게 되었는데, 다음부터는 잘 나눠보도록 하겠습니다.

KoCeleste committed:
feat: 공고상세조회API준비

- MSW핸들러및API함수추가

KoCeleste committed:
feat: 공고 상세 조회 페이지 구현 (#67)

- 공고 상세 페이지 UI 구현 (DetailHeader, DetailInfo, DetailContent)
- 공고 내용, 강의 목록, 첨부 파일 섹션 추가
- 공고 데이터 구조 개선 (Lecture, Attachment 객체화)
- 15개 공고에 lectureList, attachments 데이터 추가
- 상세 조회 및 조회수 증가 API 추가
- MSW 핸들러 구현 (GET, POST)
- 라우팅 및 ProtectedRoute 추가
- RecommendedCard, RecruitmentCard 렌더링 버그 수정

KoCeleste committed:
chore: zod 라이브러리추가
- 공고 상세 페이지 데이터 유효성 검증을 위해 추가

agrade1 commented on src/api/axios.ts:
```diff
@@ -29,7 +29,17 @@ export const getRecruitments = async (params: {
   if (params.sort) queryParams.append('sort', params.sort)
 
   const response = await axiosInstance.get(
```
이 부분에 대한건 공고 목록 코드는 api 디렉토리 하위에 폴더를 따로 만들어서 분리시켜야할거같습니다.
axios.ts 는 공통 부분에 대한 로직으로 관리하는게 좋을거같아요

그리고 api 호출 엔드포인트가 달라서 dev에서 안보이는 이슈 수정이 된거로 보이는대 해당 이슈 pr 내용에 같이 작성부탁드려요

yeeun-kor commented:
1. 이슈 목록 추가 필요 - #86
2. 충돌이슈에서 axios경로 설정은 `/v1 ` 경로가 맞는 것 같습니다.!!!

KoCeleste committed:
refactor: API 함수 분리 및 v1 경로로 수정

- axios.ts는 인스턴스만 관리
- recruitment 관련 API 함수를 src/api/recruitments.ts로 분리
- 엔드포인트를 /v1/recruitments로 수정하여 dev 환경 이슈 해결

KoCeleste committed:
Merge branch 'develop' into feat/67-RecruitmentDetail

KoCeleste committed:
chore: RecruitmentDetail 브랜치 불필요 코드 정리

- axios 스터디 공고 목록 코드 제거
- 상세 페이지 인증 store 참조 수정

yeeun-kor commented on src/api/recruitments.ts:
로직 분리 확인했습니다!

yeeun-kor commented on src/components/common/ProtectedRoute.tsx:
```diff
@@ -0,0 +1,19 @@
+}
+
+function ProtectedRoute({ children }: ProtectedRouteProps) {
+  const loginState = useAuthStore((state) => state.loginState)
+  const location = useLocation()
+
+  if (loginState === 'GUEST') {
+    return <Navigate to="/login" replace state={{ from: location.pathname }} />
+  }
+
+  return children
```
전역상태의 로그인 상태 꺼내서 쓰려면
useAuthStore에서 구조분해 할당으로 상태값 꺼내면 됩니다 
예시 
```
const {loginState} = useAuthStore()
```

KoCeleste committed:
chore: 인증 라우팅 및 공고 API 구조 정리
- ProtectedRoute 기반 접근 제어로 책임 분리
- 스터디 공고 관련 API 경로 정리
